rllib/utils/replay_buffers/prioritized_episode_replay_buffer.py

```diff
@@ -376,12 +402,20 @@ def sample(
             "weights": np.array(weights),
             "n_steps": np.array(n_steps),
         }
+        # Include infos if necessary.
         if include_infos:
             ret.update(
                 {
                     SampleBatch.INFOS: np.array(infos),
                 }
             )
+        # Include extra model outputs, if necessary.
+        if include_extra_model_outputs:
+            ret.update(
+                {
+                    "extra_model_outputs": np.array(extra_model_outputs),
```

Comment:
Not sure this is a good idea just np'ing stuff like this. This often leads to these unwieldy object arrays that have unpredictable behavior (the same is true for np'ing the infos above, we should just keep them as a list of infos-dicts in the returned batch).

We usually separate these sub-columns in `extra_model_outputs` in our batches. Can we do that here, too?

```
ret.update(
{
    k: batch(v)
    for k, v in extra_model_outputs.items()
}
)
```

The final batch (returned from `sample`) should have columns at the top level, e.g. OBS or ACTION_DIST_INPUTS.
Under each of these columns should be a (possibly nested) struct of numpy array leafs (or simply a numpy array if no complex space/struct). All leafs should have the shape (B, T?, ...), where T might be 0 or 1.

Let me know, if I'm making a thinking-mistake here. :)

Reply:
Hey @sven1977 thanks for the review! Yes this was somehow still ambiguous how to deal with the extra model outputs. I can batch the items from this field such that each of the keys in `extra_model_outputs` defines a new column in the batch.

Reply:
@sven1977 following your logic above it might also make sense to keep the other "batch" columns here as lists such that they can be `batch`ed in a standard way in the connectors?